training/run_eval.py

```diff
@@ -702,23 +738,55 @@ def _forward_time(*args, **kwargs):
 
     stats_dataset = DatasetDict()
 
-    all_stats = {"rtf": 0, "wer": 0}
+    all_stats = {"rtf": 0, "wer": 0, "tokens_per_sec": 0}
     rtf_stats = {
         "times_audio_total": 0,
         "times_transcription_total": 0,
     }
 
+    def benchmark_gen_time():
+        if model_pipeline is None:
```

Comment:
There's also the case when `model_pipeline` is not None, which we haven't covered in this PR. Here, you can pass ad-hoc generation arguments as follows:
```python
# Time forward: let's make sure that only forward is timed and not pre- and post-processing
time_result = []

def _forward_time(*args, **kwargs):
    start_time = time.time()
    result = model_pipeline_forward(*args, **kwargs)
    end_time = time.time() - start_time
    time_result.append(end_time)
    return result

model_pipeline._forward = _forward_time
result = model_pipeline(
    inputs,
    batch_size=PIPELINE_BATCH_SIZE,
    generate_kwargs={"encoder_outputs": dummy_encoder_outputs, "min_new_tokens": min_new_tokens, "max_new_tokens": max_new_tokens, **gen_kwargs},
)[0]["text"]
```